src/containers/AccountModal/forms/ChooseOffer.tsx

```diff
@@ -1,109 +1,71 @@
-import { useConfigStore } from '#src/stores/ConfigStore';
 import { useCheckoutStore } from '#src/stores/CheckoutStore';
-import { getOffer } from '#src/services/checkout.service';
 import LoadingOverlay from '#src/components/LoadingOverlay/LoadingOverlay';
 import ChooseOfferForm from '#src/components/ChooseOfferForm/ChooseOfferForm';
 import useForm, { UseFormOnSubmitHandler } from '#src/hooks/useForm';
-import type { ChooseOfferFormData, OfferPeriodicity, OfferType } from '#types/account';
-import type { Offer } from '#types/checkout';
+import type { ChooseOfferFormData, OfferType } from '#types/account';
 
 const ChooseOffer = () => {
   const history = useHistory();
   const { t } = useTranslation('account');
-  const config = useConfigStore(({ config }) => config);
-  const { cleengSandbox, json } = config;
-  const { setOffer, setOfferType } = useCheckoutStore(({ setOffer, setOfferType }) => ({ setOffer, setOfferType }), shallow);
-
-  const cleengMonthlyOffer = json?.cleengMonthlyOffer as string;
-  const cleengYearlyOffer = json?.cleengYearlyOffer as string;
-
-  const { requestedMediaOffers } = useCheckoutStore(({ requestedMediaOffers }) => ({ requestedMediaOffers: requestedMediaOffers || [] }), shallow);
-
-  // `useQueries` is not strongly typed :-(
-  const { data: monthlyOfferData, isLoading: isMonthlyOfferLoading } = useQuery(['offer', cleengMonthlyOffer], () =>
-    getOffer({ offerId: cleengMonthlyOffer }, cleengSandbox),
-  );
-  const { data: yearlyOfferData, isLoading: isYearlyOfferLoading } = useQuery(['offer', cleengYearlyOffer], () =>
-    getOffer({ offerId: cleengYearlyOffer }, cleengSandbox),
-  );
-
-  const tvodOfferQueries = useQueries(
-    requestedMediaOffers?.map(({ offerId }) => ({
-      queryKey: ['offer', offerId],
-      queryFn: () => getOffer({ offerId }, cleengSandbox),
-      enabled: !!offerId,
-    })),
-  );
+  const { setOffer } = useCheckoutStore(({ setOffer }) => ({ setOffer }), shallow);
+  const { tvodOffers, yearlyOffer, monthlyOffer, hasPremierOffer, isLoadingOffers } = useOffers();
+  const allOffers = useMemo(() => [yearlyOffer, monthlyOffer, ...tvodOffers], [yearlyOffer, monthlyOffer, tvodOffers]);
+  const [offerType, updateOfferType] = useState<OfferType>('svod');
 
-  const isTvodOffersLoading = useMemo(() => tvodOfferQueries.some(({ isLoading }) => isLoading), [tvodOfferQueries]);
+  const validationSchema: SchemaOf<ChooseOfferFormData> = object().shape({
+    offerId: mixed<string>().required(t('choose_offer.field_required')),
+  });
 
-  const tvodOffers = useMemo(
-    () => tvodOfferQueries.reduce<Offer[]>((prev, cur) => (cur.isSuccess && cur.data?.responseData ? [...prev, cur.data.responseData] : prev), []),
-    [tvodOfferQueries],
-  );
-  const premierOfferId = requestedMediaOffers.some((offer) => offer.premier);
+  const initialValues: ChooseOfferFormData = {
+    offerId: yearlyOffer?.offerId || monthlyOffer?.offerId || allOffers?.[0]?.offerId,
```

Comment:
I think this logic could be better put into the useOffers hook. Return a "defaultOfferId" property. Then we don't need to return as much other detail from the hook.

Reply:
Done